packages/outstatic/src/pages/add-custom-field/index.tsx

```diff
@@ -0,0 +1,557 @@
+    customFields,
+    deleteField = false
+  }: {
+    customFields: any
+    deleteField?: boolean
+  }) => {
+    const oid = await fetchOid()
+    const customFieldsJSON = JSON.stringify(
+      {
+        title: collection,
+        type: 'object',
+        properties: { ...customFields }
+      },
+      null,
+      2
+    )
+
+    const capi = createCommitApi({
+      message: `feat(${collection}): ${
+        deleteField ? 'delete' : 'add'
+      } ${fieldName} field`,
+      owner: repoOwner || session?.user?.login || '',
+      oid: oid ?? '',
+      name: repoSlug,
+      branch: repoBranch
+    })
+
+    capi.replaceFile(
+      `${
+        monorepoPath ? monorepoPath + '/' : ''
+      }${contentPath}/${collection}/schema.json`,
+      customFieldsJSON + '\n'
+    )
+
+    const input = capi.createInput()
+
+    return await createCommit({ variables: { input } })
+  }
+
+  const onSubmit: SubmitHandler<CustomFieldForm> = async (
+    data: CustomFieldForm
+  ) => {
+    setAdding(true)
+    const { title, ...rest } = data
+    const fieldName = camelCase(title)
+
+    if (!selectedField && customFields[fieldName]) {
+      methods.setError('title', {
+        type: 'manual',
+        message: 'Field name is already taken.'
+      })
+      setAdding(false)
+      return
+    }
+
+    try {
+      customFields[fieldName] = {
+        ...rest,
+        title: capitalize(data.title)
+      }
+
+      const created = await capiHelper({ customFields })
+
+      if (created) {
+        setCustomFields({ ...customFields })
+      }
+    } catch (error) {
+      // TODO: Better error treatment
+      setError('add')
+      console.log({ error })
+    }
+
+    setHasChanges(false)
+    setSelectedField('')
+    setShowAddModal(false)
+    setAdding(false)
+  }
+
+  const deleteField = async (name: string) => {
+    setDeleting(true)
+
+    try {
+      let newCustomFields = { ...customFields }
+      delete newCustomFields[name]
+      const deleted = await capiHelper({
+        customFields: newCustomFields,
+        deleteField: true
+      })
+
+      if (deleted) {
+        setCustomFields(newCustomFields)
+      }
+    } catch (error) {
+      // TODO: Better error treatment
+      setError('delete')
+      console.log({ error })
+    }
+    setSelectedField('')
+    setShowDeleteModal(false)
+    setDeleting(false)
+    setHasChanges(false)
+  }
+
+  useEffect(() => {
+    const documentQueryObject = schemaQueryData?.repository?.object
+    if (documentQueryObject?.__typename === 'Blob') {
+      const schema = JSON.parse(documentQueryObject?.text || '{}')
+      setCustomFields(schema.properties)
+    }
+  }, [schemaQueryData])
+
+  useEffect(() => {
+    const subscription = methods.watch(() => setHasChanges(true))
+    return () => subscription.unsubscribe()
+  }, [methods])
+
+  // Ask for confirmation before leaving page if changes were made.
+  useNavigationLock(hasChanges)
+
+  return (
+    <FormProvider {...methods}>
```

Comment:
`<AdminLayout>` should be the top component, just to ensure things don't leak outside of the nav & footer containers.